Wrappers/Python/cil/processors/MaskGenerator.py

```diff
@@ -20,8 +20,100 @@
 import numpy
 from scipy import special, ndimage
 
+class MaskGenerator(object):
+    """
+    This class contains factory methods to create a MaskGenerator object using the desired algorithm.
+    """
+    @staticmethod
+    def special_values(nan=True, inf=True):
+        r'''This creates a MaskGenerator processor which generates a mask for inf and/or nan values.
 
-class MaskGenerator(DataProcessor):
+        :param nan: mask NaN values
+        :type nan: bool, default=True
+        :param inf: mask INF values
+        :type inf: bool, default=True
+
+        '''
+        if nan is True:
+            if inf is True:
+                proccessor = MaskGenerator_backend(mode='special_values')
```

Comment:
why do you need to create `MaskGenerator_backend`? Can't these methods be static methods of `MaskGenerator`.